ngraph/test/function.cpp

```diff
@@ -511,3 +512,243 @@ TEST(function, DISABLED_create_function_with_incorrect_tensor_names_from_const_f
+    {
+        ov::PartialShape shape({1, 3, 22, 22});
+        ov::element::Type type(ov::element::Type_t::f32);
+        auto param = std::make_shared<ov::op::v0::Parameter>(type, shape);
+        param->get_output_tensor(0).set_names({"tensor"});
+        auto relu = std::make_shared<ov::op::v0::Relu>(param);
+        auto result = std::make_shared<ov::op::v0::Result>(relu);
+
+        ov::ParameterVector params = {param};
+        ov::ResultVector results = {result};
+
+        ngraph = std::make_shared<ov::Function>(results, params);
+    }
+
+    ASSERT_EQ(ngraph->get_parameters()[0]->get_shape(), ov::Shape({1, 3, 22, 22}));
+    ASSERT_EQ(ngraph->get_results()[0]->get_shape(), ov::Shape({1, 3, 22, 22}));
+
+    {
+        std::map<std::string, ov::PartialShape> new_shape;
+        new_shape["tensor"] = refShape;
+        ASSERT_NO_THROW(ngraph->reshape(new_shape));
+    }
+
+    ASSERT_TRUE(ngraph->input(0).get_partial_shape().is_dynamic());
+    ASSERT_TRUE(ngraph->output(0).get_partial_shape().is_dynamic());
+    ASSERT_EQ(ngraph->input(0).get_partial_shape(), refShape);
+    ASSERT_EQ(ngraph->output(0).get_partial_shape(), refShape);
+}
+
+TEST(function_reshape, ReshapeSpatialReLUStaticToFullyDynamic) {
+    const ov::PartialShape refShape = ov::PartialShape::dynamic();
+    std::shared_ptr<ov::Function> ngraph;
+    {
+        ov::PartialShape shape({1, 3, 22, 22});
+        ov::element::Type type(ov::element::Type_t::f32);
+        auto param = std::make_shared<ov::op::v0::Parameter>(type, shape);
+        param->get_output_tensor(0).set_names({"tensor"});
+        auto relu = std::make_shared<ov::op::v0::Relu>(param);
+        auto result = std::make_shared<ov::op::v0::Result>(relu);
+
+        ov::ParameterVector params = {param};
+        ov::ResultVector results = {result};
+
+        ngraph = std::make_shared<ov::Function>(results, params);
+    }
+
+    ASSERT_EQ(ngraph->get_parameters()[0]->get_shape(), ov::Shape({1, 3, 22, 22}));
+    ASSERT_EQ(ngraph->get_results()[0]->get_shape(), ov::Shape({1, 3, 22, 22}));
+
+    {
+        std::map<std::string, ov::PartialShape> new_shape;
+        new_shape["tensor"] = refShape;
+        ASSERT_NO_THROW(ngraph->reshape(new_shape));
+    }
+
+    ASSERT_TRUE(ngraph->input().get_partial_shape().is_dynamic());
+    ASSERT_TRUE(ngraph->output().get_partial_shape().is_dynamic());
+    ASSERT_EQ(ngraph->input().get_partial_shape(), refShape);
+    ASSERT_EQ(ngraph->output().get_partial_shape(), refShape);
+}
+
+TEST(function_reshape, ReshapeSpatialReLUDynamicToDynamic) {
+    const ov::PartialShape refShape{1, 3, ov::Dimension::dynamic(), 25};
+    std::shared_ptr<ov::Function> ngraph;
+    {
+        ov::PartialShape shape({1, 3, 22, ov::Dimension::dynamic()});
+        ov::element::Type type(ov::element::Type_t::f32);
+        auto param = std::make_shared<ov::op::v0::Parameter>(type, shape);
+        param->get_output_tensor(0).set_names({"tensor"});
```

Comment:
Propose to add next test:
```suggestion
        param->get_output_tensor(0).set_names({"t1", "t2"});
        function->reshape({"t1": {1, 3, 3, 3}, "t2": {1,  3, 2, 5}});
```

Reply:
added here https://github.com/openvinotoolkit/openvino/pull/7776